scripts/build-layout-scales.js

```diff
@@ -0,0 +1,1271 @@
+    /**
+      * Set an element's minimum width.
+      *
+      * @group
+      * @memberof Sizing
+      * @example
+      * <div class='inline-block wmin96 bg-darken10'>wmin96</div>
+     */`
+    );
+
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .wmin${scale} {
+          min-width: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .wmin${scale}-mm {
+          min-width: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .wmin${scale}-ml {
+          min-width: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--xl-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .wmin${scale}-mxl {
+          min-width: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    return css;
+  };
+
+  variantGenerators.height = function (scales) {
+    function value(v) {
+      return v === '-full' ? '100%' : `${v}px`;
+    }
+
+    let css = stripIndent(`
+    /**
+      * Set an element's height.
+      *
+      * @group
+      * @memberof Sizing
+      * @example
+      * <div class='h24 bg-darken10'>h24</div>
+     */`
+    );
+
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .h${scale} {
+          height: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += '/** @endgroup */\n';
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--m-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .h${scale}-mm {
+          height: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--l-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .h${scale}-ml {
+          height: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    css += stripIndent(`\n\n@media ${mediaQueries['--xl-screen']} {`);
+    css += scales.reduce((result, scale) => {
+      return result += stripIndent(`
+        .h${scale}-mxl {
+          height: ${value(scale)} !important;
+        }
+      `);
+    }, '');
+    css += stripIndent('}');
+
+    return css;
+  };
+
+  variantGenerators.maxHeight = function (scales) {
+    function value(v) {
+      return v === '-full' ? '100%' : `${v}px`;
```

Comment:
There should be a shared helper that translates entries into CSS values that is used by every one of these, right? `-full` always means `100%`, `-neg{n}` always means `-{n}px`.